### DISCUSSION TITLE
dagster-databricks databricks SDK version mismatch

### DISCUSSION THREAD

**Discussion by geoHeil** [dagster-io/dagster#18422] [Q&A] [answered]
Following up on what I see in https://dagster.slack.com/archives/C01U954MEER/p1701357986154769

> Hi, I'm trying to submit a databricks job using dagster, I'm getting an unexpected error: module 'databricks.sdk.service.jobs' has no attribute 'JobsHealthRule'

After deploying a 14.2 databricks cluster I started to take a look for myself and observe

-  OSS https://pypi.org/project/databricks-sdk/ is 0.14.0 
 - dagster https://github.com/dagster-io/dagster/blob/master/python_modules/libraries/dagster-databricks/setup.py#L41C10-L41C26 still on <0.9  
-  databricks itself is on 0.1.6 (see the code below)

I checked this with:

```
from importlib import metadata

try:
    version = metadata.version("databricks-sdk")
    print(f"databricks-sdk version: {version}")
except metadata.PackageNotFoundError:
    print("databricks-sdk is not installed")

```

What versions are the right ones? Is there the need for exact matches?

Very strange is when executing:

```
pip install databricks_sdk==0.14
```

for both the client (server where dagster submits the task from the step launcher) and the server (the databricks notebook as well I still keep getting:

```
'databricks.sdk.service.jobs' has no attribute 'JobsHealthRule'
```

In fact even databricks themselves is listing https://docs.databricks.com/en/release-notes/runtime/14.2.html `databricks-sdk 0.1.6` 

Interestingly, when using 0.1.6 (matching) both locally and on the server Dagster even fails to load the codelocaiton with that mentioned error message of a not fitting class

**Comment by HPicatto** [comment]
dagster-databricks, in it's most recent version, depends on databricks-sdk<0.9. so the cluster must be adapted to that version

**Comment by geoHeil** [comment, marked as answer]
```

{"pypi": {"package": "databricks-sdk<0.9"}},

```

when adding this - in fact a more recent verssion like 0.14.0 seems to be OK as well there are 2 options:

- databricks-sdk<0.9 is pre-added to the launched cluster then the job will work
- it is not and the job will first fail, then install it and then a 2nd run will work

This is in the case of an already running cluster

if dagster is starting the cluster and that library is to be installed in a suitably new version it will work

**Reply by geoHeil** [reply]
https://github.com/dagster-io/dagster/issues/18428 for fixing this
